src/tokenizer/ngram_tokenizer.rs

```diff
@@ -0,0 +1,121 @@
+use super::{Token, TokenStream, Tokenizer};
+
+///Tokenize the text by splitting words into ngrams of the given size
+#[derive(Clone)]
+pub struct NgramTokenizer {
+  min_gram: usize,
+  max_gram: usize,
+  edges_only: bool,
+}
+
+impl NgramTokenizer {
+  /// Configures a new Ngram tokenizer
+  pub fn new(min_gram: usize, max_gram: usize, edges_only: bool) -> NgramTokenizer {
+    assert!(min_gram > 0, "min_gram must be greater than 0");
+    assert!(
+      min_gram <= max_gram,
+      "min_gram must not be greater than max_gram"
+    );
+
+    NgramTokenizer {
+      min_gram,
+      max_gram,
+      edges_only,
+    }
+  }
+}
+pub struct NgramTokenStream<'a> {
+  text: &'a str,
+  location: usize,
+  text_length: usize,
+  token: Token,
+  min_gram: usize,
+  max_gram: usize,
+  gram_size: usize,
+  edges_only: bool,
+}
+
+impl<'a> Tokenizer<'a> for NgramTokenizer {
+  type TokenStreamImpl = NgramTokenStream<'a>;
+
+  fn token_stream(&self, text: &'a str) -> Self::TokenStreamImpl {
+    NgramTokenStream {
+      text,
+      location: 0,
```

Comment:
For consistency with the rest of tantivy naming, maybe rename `location`, `cursor` or `offset_cursor` or `token_start`

Reply:
Will do, but one question. I chose location so it would match the `JapaneseTokenizer` and the `RawTokenizer`. Would you still like it to be changed?

Reply:
I don't see `location` in the `JapaneseTokenizer` nor the `RawTokenizer`.